qiime2/sdk/usage.py

```diff
@@ -1241,25 +1238,31 @@ def construct_collection(
         ... )
         >>> mapping_1
         <ExecutionUsageVariable name='mapping_1', var_type='artifact'>
-        >>> collection_1 = use.construct_collection(
-        ...     'collection_1', 'artifact', {'a': mapping_1, 'b': mapping_1}
+        >>> collection_1 = use.construct_artifact_collection(
+        ...     'collection_1', {'a': mapping_1, 'b': mapping_1}
         ... )
         >>> collection_1
         <ExecutionUsageVariable name='collection_1', var_type='artifact_collection'>
         '''  # noqa: E501
-        if member_type not in ('artifact', 'visualization'):
-            msg = (
-                'Collections must be either of type artifact or of type '
-                f'visualization, not {member_type}.'
-            )
-            raise ValueError(msg)
+
+        # make sure members is dict to avoid repeated type checking
+        if type(members) is list:
+            members = {str(i): member for i, member in enumerate(members)}
 
         if not all(
-            member.var_type == member_type for member in members.values()
+            member.var_type == 'artifact' for member in members.values()
         ):
+            raise ValueError('Expected only artifacts in the collection.')
+
+        str_ns = {str(name) for name in self.namespace}
+        diff = set(
+            str(member.to_interface_name()) for member in members.values()
+        ) - str_ns
```

Comment:
```suggestion
        diff = set(
            str(member.to_interface_name()) for member in members.values()
        ) - self.namespace
```

Reply:
It's actually lying about `self.namespace` being a set of strings. It's actually a set of `<class 'qiime2.plugins.ArtifactAPIUsageVariable.repr_raw_variable_name'>
`, so if we don't cast it to str the diff always exists unless both sets are empty.

Reply:
GROSS. Ok, thank you good deal then.